yaptide/celery/tasks.py

```diff
@@ -257,10 +270,16 @@ def merge_results(results: list[dict]) -> dict:
 
         if averaged_estimators is None:
             averaged_estimators: list[dict] = result.get("estimators", [])
+            total_particles += int(averaged_estimators[0]["metadata"]["number_of_primaries"])
+            logging.debug(total_particles)
             # There is nothing to average yet
             continue
 
-        averaged_estimators = average_estimators(averaged_estimators, result.get("estimators", []), i)
+        new_particles = int(result.get("estimators", [])[0]["metadata"]["number_of_primaries"])
+        logging.debug("Avereging results from %d particles", new_particles)
+        averaged_estimators = average_estimators(averaged_estimators, result.get("estimators", []), total_particles,
```

Comment:
does it calculate normal average or weighted average ?
in case of interuppted simulation a weighted average would be needed.
Maybe you can (re)use this code: https://github.com/DataMedSci/pymchelper/blob/master/pymchelper/averaging.py

Reply:
It calculates weighted average

Reply:
But not all quantities should be averaged, see details on pymchelper code.
An example pymchelper code usage can be found here:
https://github.com/yaptide/yaptide/blob/7a66b1c1d846987f380d6b5ba5d0ab6d7b9e31a0/yaptide/celery/utils/pymc.py#L111

As for weighted average - do you mean this code is doing weighted average ?
https://github.com/yaptide/yaptide/blob/7a66b1c1d846987f380d6b5ba5d0ab6d7b9e31a0/yaptide/celery/utils/pymc.py#L172-L174

If thats the case, then its completely unreadable. It would need a good paragraph of docstring explaining that. 

Ideally accompanied with doctest: https://docs.python.org/3/library/doctest.html

Reply:
It was doing a weighted average. maybe it was overwritten in some merge from master

Reply:
As for now it's ok, but the code is completely unreadable. It's hard to guess from reading the code and comments what it is doing.